deduplicator/src/lib/deduplicator.rs

```diff
@@ -90,7 +90,7 @@ impl Deduplicator {
         );
 
         let conn_get_collisions = self.db.get_conn()?;
-        let mut sorted_hashes = DbHashes::get_sorted_hashes(&conn_get_collisions)?;
+        let mut sorted_hashes = DbHashes::get_collisions_iter(&conn_get_collisions)?;
```

Comment:
I just wondered: how is this iterator working under the table? Querying data by subset (most likely) or collecting all of them at once? If this is the first, wouldn't collect data at once be faster? (This comment is very theorical and based on non-verified assumptions.)

Reply:
This is actually just a wrapper around `rusqlite`'s iterator, yielding parsed data to make the implementation transparent. I'm not sure it would really be faster to load all data at once since this is the only IO we perform. Plus, it would be really expensive in memory to do so :/

Reply:
I guess we could experiment using a buffer, but if I recall correctly, we would still have to parse data into something to get ownership of it :(

Reply:
So not ideal either... It might be worth keeping it in mind though. I don't expect big performance improvement but even one second on a small subset might end up in something interesting at a bigger level.